simtools/util/general.py

```diff
@@ -770,3 +776,93 @@ def get_log_excerpt(log_file, n_last_lines=30):
+
+    Returns
+    -------
+    np.array
+        The values of the 1D histogram with size = int(max_dist/bin_size).
+    np.array
+        The edges of the 1D histogram with size = int(max_dist/bin_size) + 1.
+
+    """
+    logger = logging.getLogger(__name__)
+    # Check if the histogram will make sense
+    bins_step = 2 * max_dist / bins  # in the 2D array, the positive and negative direction count.
+    for axis in [xaxis, yaxis]:
+        if (bins_step < np.diff(axis)).any():
+            msg = (
+                f"The histogram with number of bins {bins} and maximum distance of {max_dist} "
+                f"resulted in a bin size smaller than the original array. Please adjust those "
+                f"parameters to increase the bin size and avoid nan in the histogram values."
+            )
+            logger.warning(msg)
+            break
+
+    grid_2d_x, grid_2d_y = np.meshgrid(xaxis[:-1], yaxis[:-1])  # [:-1], since xaxis and yaxis are
+    # the hist edges (n + 1).
+    # radial_distance_map maps the distance to the center from each element in a square matrix.
+    radial_distance_map = np.sqrt(grid_2d_x**2 + grid_2d_y**2)
+    # The sorting and unravel_index give us the two indices for the position of the sorted element
+    # in the original 2d matrix
+    x_indices_sorted, y_indices_sorted = np.unravel_index(
+        np.argsort(radial_distance_map, axis=None), np.shape(radial_distance_map)
+    )
+
+    # We construct a 1D array with the histogram counts sorted according to the distance to the
+    # center.
+    hist_sorted = np.array(
+        [hist2d[i_x, i_y] for i_x, i_y in zip(x_indices_sorted, y_indices_sorted)]
+    )
+    distance_sorted = np.sort(radial_distance_map, axis=None)
+
+    # For larger distances, we have more elements in a slice 'dr' in radius, hence, we need to
+    # acount for it using weights below.
+
+    weights, radial_edges = np.histogram(distance_sorted, bins=bins, range=(0, max_dist))
+    histogram_1D = np.empty_like(weights, dtype=float)
+
+    for i_radial, _ in enumerate(radial_edges[:-1]):
+
+        # Here we sum all the events within a radial interval 'dr' and then divide by the number of
+        # bins that fit this interval.
+        indices_to_sum = (distance_sorted >= radial_edges[i_radial]) * (
+            distance_sorted < radial_edges[i_radial + 1]
+        )
+        histogram_1D[i_radial] = np.sum(hist_sorted[indices_to_sum]) / weights[i_radial]
+    return histogram_1D, radial_edges
+
+
+def save_dict_to_file(dictionary, file_name):
+    """
+    Save dictionary to a file.
+
+    Parameters
+    ----------
+    dictionary: dict
+        Dictionary to be saved into a file.
+    file_name: str or Path
+        Name of file to be saved with path.
+    """
+    logger = logging.getLogger(__name__)
+    file_name = Path(file_name).with_suffix(".yml")
+    logger.info(f"Exporting histogram configuration to {file_name}")
+    with open(file_name, "w") as file:
```

Comment:
No try and except when writing a file?

Reply:
I think I removed it at some point (or really forgot). Isn't sufficient that yaml raises the error if the file does not exist?
If you prefer I can also implement it here

Reply:
Only if the yaml error message is understandable. In general, python packages have very user-unfriendly error messages and are hard to parse. This is why I would prefer a clearer error message from us (but maybe here it is ok, not sure).

Reply:
Ok, I implemented a quick check and the test for it